excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/validation/ChallengeQuestionValidator.java

```diff
@@ -0,0 +1,212 @@
+import uk.gov.hmcts.ccd.definition.store.repository.entity.*;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+@Component
+public class ChallengeQuestionValidator {
+
+    private ParseContext parseContext;
+    private static final String ANSWER_MAIN_SEPARATOR = ",";
+    private static final String ANSWER_FIELD_SEPARATOR = "|";
+    private static final String ANSWER_FIELD_DOT_SEPARATOR = ".";
+    private static final String ANSWER_FIELD_ROLE_SEPARATOR = ":";
+    private static final String ANSWER_FIELD_MATCHER = "^\\$\\{\\S.{1,}.\\S.{1,}}$|^\\$\\{\\S.{1,}.\\S.{1,}}:\\[\\S{1,}\\]$";
+    private static final String ERROR_MESSAGE = "ChallengeQuestionTab Invalid";
+    private final ChallengeQuestionTabService challengeQuestionTabService;
+    private ChallengeQuestionDisplayContextParameterValidator challengeQuestionDisplayContextParameterValidator;
+
+    @Autowired
+    public ChallengeQuestionValidator(ChallengeQuestionTabService challengeQuestionTabService,
+                                      ChallengeQuestionDisplayContextParameterValidator challengeQuestionDisplayContextParameterValidator) {
+
+        this.challengeQuestionTabService = challengeQuestionTabService;
+        this.challengeQuestionDisplayContextParameterValidator = challengeQuestionDisplayContextParameterValidator;
+    }
+
+    public ChallengeQuestionTabEntity validate(ParseContext parseContext, DefinitionDataItem definitionDataItem) {
+
+        this.parseContext = parseContext;
+        validateQuestionId(definitionDataItem);
+        final String questionId = definitionDataItem.getString(ColumnName.CHALLENGE_QUESTION_QUESTION_ID);
+        Optional<ChallengeQuestionTabEntity> result = challengeQuestionTabService.findByQuestionId(questionId);
+        final ChallengeQuestionTabEntity challengeQuestionTabEntity = result.orElseGet(ChallengeQuestionTabEntity::new);
+        challengeQuestionTabEntity.setQuestionId(questionId);
+        challengeQuestionTabEntity.setCaseType(getCaseTypeEntity(definitionDataItem));
+        challengeQuestionTabEntity.setAnswerFieldType(getFieldTypeEntity(definitionDataItem));
+        validateDisplayOrder(definitionDataItem, challengeQuestionTabEntity);
+        validateQuestionText(definitionDataItem, challengeQuestionTabEntity);
+        validateID(definitionDataItem, challengeQuestionTabEntity);
+        validateAnswer(definitionDataItem, challengeQuestionTabEntity);
+        validateDisplayContext(definitionDataItem, challengeQuestionTabEntity);
+        return challengeQuestionTabEntity;
+    }
+
+    private void validateQuestionId(DefinitionDataItem definitionDataItem) {
+        final String questionId = definitionDataItem.getString(ColumnName.CHALLENGE_QUESTION_QUESTION_ID);
+        validateNullValue(questionId, ERROR_MESSAGE + " value: questionId cannot be null.");
```

Comment:
can we accumulate all the errors instead of throwing back the individual errors...